### PR TITLE
Gauss quadrature subsampling

### PR DESCRIPTION
**Pull request by akilnarayan**
New features and syntax

- separated sampling from training
- PolynomialChaosExpansion accepts
  + list of distributions as input
  + integer order parameter for TotalDegreeIndexing
- new sampling methods
  + tensor-product Gauss quadrature
  + induced sampling from a tensorized GQ mesh
- new visualization tools. See `demos/build_pce.py` for a demo

### CODE REVIEW

File: UncertainSCI/distributions.py [hunk 1]
```diff
@@ -743,7 +749,8 @@ def _detect_dimension(self, alpha, beta, dim, domain):
 
             return
 
-        elif domain is not None and domain.shape[1] > 1:  # Case 3
+        #elif domain is not None and domain.shape[1] > 1:  # Case 3
```
Comment (dcwhite) [inline UncertainSCI/distributions.py:752]:
```suggestion

```

File: UncertainSCI/pce.py [hunk 2]
```diff
@@ -155,46 +246,33 @@ def integration_weights(self):
...
     def build_pce_wlsq(self):
         """
         Performs a (weighted) least squares PCE surrogate using saved samples
         and model output.
         """
 
-        p_standard = self.distribution.transform_standard_dist_to_poly.map(
-                    self.distribution.transform_to_standard.map(self.samples))
+        #p_standard = self.distribution.transform_standard_dist_to_poly.map(
+        #            self.distribution.transform_to_standard.map(self.samples))
```
Comment (dcwhite) [inline UncertainSCI/pce.py:267]:
```suggestion
```

File: tests/test_pce.py [hunk 3]
```diff
@@ -39,7 +39,9 @@ def mymodel(p):
         indices = TotalDegreeSet(dim=1, order=3)
         pce = PolynomialChaosExpansion(indices, dist)
 
-        lsq_residuals = pce.build(mymodel, fast_sampler=False)
+        pce.sampling_options = {'fast_sampler': False}
+        # lsq_residuals = pce.build(mymodel, fast_sampler=False)
```
Comment (dcwhite) [inline tests/test_pce.py:43]:
```suggestion
```

**Review by dcwhite — COMMENTED** [review]
Looks great! The suggestions all just remove commented-out code.

**Comment by akilnarayan** [comment]
Ok, those fixes are great. I committed. Let me know if there are other issues.

**Review by jessdtate — APPROVED** [review]

**Pull request merged**